executor.py

```diff
@@ -1,7 +1,143 @@
+         value for `mode` is db4 - Daubechies wavelets.
+         :param is_embed_bool: Set to `True` to encode the images into boolean embeddings using the hashing technique.
+         By default set to `False` to encode the images as `np.uint8` embeddings values.
+         :param traversal_paths: The default traversal path on docs, e.g. ['r'], ['c']
+        """
+        super().__init__(**kwargs)
+        if not hash_type or hash_type not in HASH_TYPE:
+            raise ValueError('Please select one of the `hash_type`')
+        self.hash_type = hash_type
+        self.hash_size = hash_size
+
+        self._average_hash_args = average_hash_args or {}
+        if 'avg_pixels' in self._average_hash_args:
+            if self._average_hash_args['avg_pixels'] == 'mean':
+                self._average_hash_args['mean'] = np.mean
+            else:
+                self._average_hash_args['mean'] = np.median
+            del self._average_hash_args['avg_pixels']
+
+        self._perceptual_hash_args = perceptual_hash_args or {}
+        self._perceptual_hash_args.setdefault('highfreq_factor', 4)
+
+        self._wavelet_hash_args = wavelet_hash_args or {}
+        self._wavelet_hash_args.setdefault('image_scale', None)
+        self._wavelet_hash_args.setdefault('mode', 'haar')
+
+        self.is_embed_bool = is_embed_bool
+        self.traversal_paths = traversal_paths
+        self.logger = JinaLogger(
+            getattr(self.metas, 'name', self.__class__.__name__)
+        ).logger
+
     @requests
-    def foo(self, docs: DocumentArray, **kwargs):
-        pass
+    def encode(self, docs: DocumentArray, parameters: Dict = {}, **kwargs):
+        """
+        Read the numpy arrays from the Document.blob, and encode it into an embeddings using the
+        comparable hashing technique. Features in the image are used to generate a distinct
+        (but not unique) fingerprint, and these fingerprints are comparable.
+
+        :param docs: A document array with documents to create embeddings for. Only the
+            documents that have the ``blob`` attribute will get embeddings. The ``blob``
+            attribute should be the numpy array of the image, and should have dtype
+            ``np.uint8``
+        :param parameters: A dictionary that contains parameters to control encoding.
+            The accepted keys are: `perceptual_hash_args`, `average_hash_args`, `wavelet_hash_args`,
+            and `traversal_paths`.
+            For example: `parameters={'average_hash_args': {'avg_pixels': 'median'}}`
+        """
+        for doc in docs.traverse_flat(
+            parameters.get('traversal_paths', self.traversal_paths)
+        ):
+            if doc.blob is None:
+                self.logger.error(f'No blob passed for the Document with id: {doc.id}')
+                continue
+
+            image = Image.fromarray(doc.blob)
+            hash_hex = None
+            try:
+                if self.hash_type == 'wavelet':
```

Comment:
Actually, do we need to initialize the Executor with a predefined `self.hash_type`? Can't this be ad-hoc, per request?